pravega/pravega
# Issue 5523: Enabling GithubActions and Disabling Travis CI

**Change log description**  
- Disabled .travis.yml as we will no longer be using Travis CI. This file will be deleted in a subsequent PR.
- Set up GitHub Actions workflow as follows:
   - Triggers on master branch and release branch pushes, on every pull request (regardless of branch) and release publish.
   - Has a compile and static analysis job, followed by a few parallel test jobs, then a check and a snapshot (only for master branch and release branches).
   - Check job is called `CI Complete`.
- Bonus features: 
    - Build output is cached after the `Build` step and reused for subsequent jobs. This speeds up the overall workflow run.
    - Test execution output, history and reports is cached on a per workflow-job basis. If a workflow is re-run for whatever reason (such as one job failing), the successful jobs will execute much faster (by skipping all tests that passed).

**Purpose of the change**  
Fixes #5523.
Fixes #5538.

**What the code does**  
See description.

**How to verify it**  
Please verify that this is functionally equivalent to the previous CI pipeline:
- It compiles the code.
- It performs static analysis.
- It executes all unit tests and all integration tests.
- It uploads code coverage correctly.
- For master and release branches, it publishes to JFrog artifactory.
- Triggers are correct and make sense.
